app/controllers/api/users.php

```diff
@@ -376,6 +377,131 @@
+    ->label('sdk.description', '/docs/references/users/update-user-password.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_USER)
+    ->param('userId', '', new UID(), 'User unique ID.')
+    ->param('password', '', new Password(), 'New user password. Must be between 6 to 32 chars.')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('audits')
+    ->action(function ($userId, $password, $response, $dbForInternal, $audits) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Event\Event $audits */
+
+        $user = $dbForInternal->getDocument('users', $userId);
+
+        if ($user->isEmpty()) {
+            throw new Exception('User not found', 404);
+        }
+
+        $user = $dbForInternal->updateDocument('users', $user->getId(), $user->setAttribute('password', Auth::passwordHash($password))
+            ->setAttribute('passwordUpdate', \time()));
+
+        $audits
+            ->setParam('userId', $user->getId())
+            ->setParam('event', 'users.update.password')
+            ->setParam('resource', 'users/'.$user->getId())
+        ;
+
+        $response->dynamic($user, Response::MODEL_USER);
+    });
+
+App::patch('/v1/users/:userId/email')
+    ->desc('Update Email')
+    ->groups(['api', 'users'])
+    ->label('event', 'users.update.email')
+    ->label('scope', 'users.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'users')
+    ->label('sdk.method', 'updateEmail')
+    ->label('sdk.description', '/docs/references/users/update-user-email.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_USER)
+    ->param('userId', '', new UID(), 'User unique ID.')
+    ->param('email', '', new Email(), 'User email.')
+    ->inject('response')
+    ->inject('dbForInternal')
+    ->inject('audits')
+    ->action(function ($userId, $email, $response, $dbForInternal, $audits) {
+        /** @var Appwrite\Utopia\Response $response */
+        /** @var Utopia\Database\Database $dbForInternal */
+        /** @var Appwrite\Event\Event $audits */
+
+        $user = $dbForInternal->getDocument('users', $userId);
+
+        if ($user->isEmpty()) {
+            throw new Exception('User not found', 404);
+        }
+
+        $email = \strtolower($email);
+        $profile = $dbForInternal->findOne('users', [new Query('email', Query::TYPE_EQUAL, [\strtolower($email)])]); // Get user by email address
+
+        if ($profile) {
+            throw new Exception('Email already exists', 409);
+        }
+
+        $user = $dbForInternal->updateDocument('users', $user->getId(), $user->setAttribute('email', $email));
```

Comment:
We can try - catch for `Duplicate` exception instead of checking it before hand. Some good examples can be found in the DB controller.

Reply:
Should I update the `account` service in similar way as well?

Reply:
Yeh, sure